src/Pickles/Pickles/DirectoryCrawler/FeatureNodeFactory.cs

```diff
@@ -57,12 +58,7 @@ public INode Create(FileSystemInfoBase root, FileSystemInfoBase location)
             if (this.relevantFileDetector.IsFeatureFile(file))
             {
                 Feature feature = this.featureParser.Parse(file.FullName);
-                if (feature != null)
-                {
-                    return new FeatureNode(file, relativePathFromRoot, feature);
-                }
-
-                throw new InvalidOperationException("This feature file could not be read and will be excluded");
+                return feature != null ? new FeatureNode(file, relativePathFromRoot, feature) : null;
```

Comment:
Sorry, but default value check is not the solution.
I think InvalidOperationException is useless because if we have an parsing error we already have an exception for malformed file in function `Parse(string filename)`  of FeatureParser.cs